src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java

```diff
@@ -265,6 +266,93 @@ public void testGenerateFeedbackSessionEmails() throws IOException {
     }
 
     @Test
+    public void testGenerateFeedbackSessionEmails_testSanitization() throws IOException {
+
+        FeedbackSessionAttributes session = fsLogic.getFeedbackSession("Normal feedback session name",
+                                                                       "idOfTestingSanitizationCourse");
+        CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+        StudentAttributes student1 = studentsLogic.getStudentForEmail(course.getId(), "normal@sanitization.tmt");
+        InstructorAttributes instructor1 =
+                instructorsLogic.getInstructorForEmail(course.getId(), "instructor1@sanitization.tmt");
+
+        ______TS("feedback session opening emails: sanitization required");
+
+        List<EmailWrapper> emails = new EmailGenerator().generateFeedbackSessionOpeningEmails(session);
+
+        assertEquals(2, emails.size());
+
+        String subject = String.format(EmailType.FEEDBACK_OPENING.getSubject(),
+                course.getName(), session.getFeedbackSessionName());
+
+        boolean hasStudent1ReceivedEmailCorrectly = false;
+        boolean hasInstructor1ReceivedEmailCorrectly = false;
+        for (EmailWrapper email : emails) {
+            if (email.getRecipient().equals(student1.email)) {
+                verifyEmail(email, student1.email, subject,
+                        "/sessionOpeningEmailTestingSanitzationForStudent.html");
+                hasStudent1ReceivedEmailCorrectly = true;
+            } else if (email.getRecipient().equals(instructor1.email)) {
+                verifyEmail(email, instructor1.email, subject,
+                        "/sessionOpeningEmailTestingSanitizationForInstructor.html");
+                hasInstructor1ReceivedEmailCorrectly = true;
+            }
+        }
+        assertTrue(hasStudent1ReceivedEmailCorrectly && hasInstructor1ReceivedEmailCorrectly);
+
+        ______TS("feedback session closed alerts: sanitization required");
+
+        emails = new EmailGenerator().generateFeedbackSessionClosedEmails(session);
+        assertEquals(2, emails.size());
+
+        subject = String.format(EmailType.FEEDBACK_CLOSED.getSubject(),
+                course.getName(), session.getFeedbackSessionName());
+
+        hasStudent1ReceivedEmailCorrectly = false;
+        hasInstructor1ReceivedEmailCorrectly = false;
+        for (EmailWrapper email : emails) {
+            if (email.getRecipient().equals(student1.email)) {
+                verifyEmail(email, student1.email, subject,
+                        "/sessionClosedEmailTestingSanitizationForStudent.html");
+                hasStudent1ReceivedEmailCorrectly = true;
+            } else if (email.getRecipient().equals(instructor1.email)) {
+                verifyEmail(email, instructor1.email, subject,
+                        "/sessionClosedEmailTestingSanitizationForInstructor.html");
+                hasInstructor1ReceivedEmailCorrectly = true;
+            }
+        }
```

Comment:
There is a very similar for-loop a few lines up. Any chance of extracting as a utility method?